### ISSUE TITLE
Cannot rename enum fields using VSCode extension.

### ISSUE THREAD

**Issue by smallkirby** [zigtools/zls#2042]
### Zig Version

0.13.0

### ZLS Version

0.13.0

### Client / Code Editor / Extensions

Visual Studio Code-1.93.1

### Steps to Reproduce and Observed Behavior

Init new project with `zig init`.
Edit `main.zig` as follows:
```zig
const std = @import("std");

pub const SomeEnum = enum {
    zero,
    one,
};

pub fn main() !void {
    const se = SomeEnum.one;
    std.log.debug("se: {?}", .{se});
}
```

Right click (or Ctrl+R) the field of enum `SomeEnum`, for example `zero`, to rename. Enter new name and press enter. Popup appears at the right bottom of VSCode saying that `Rename failed to apply edits`.

Note that I can rename struct field.

### Expected Behavior

Enum field itself and all the usage of the field are renamed.

### Relevant log output

No errors were shown in the log:

```shell
info : ( main ): Starting ZLS 0.13.0 @ '/usr/bin/zls'
info : (server): Client is 'Visual Studio Code-1.93.1'
info : (server): No config file zls.json found. This is not an error.
warning: (config): failed to open entry in PATH '/usr/bin/ccache/zig': error.NotDir
info : (server): Set config option 'builtin_path' to '/home/user/.cache/zls/builtin.zig'
info : (server): Set config option 'zig_lib_path' to '/home/user/.zvm/0.13.0/lib'
info : (server): Set config option 'zig_exe_path' to '/home/user/.zvm/bin/zig'
info : (server): Set config option 'build_runner_path' to '/home/user/.cache/zls/build_runner/21872970afd69e48a0847077e5196711/build_runner.zig'
info : (server): Set config option 'global_cache_path' to '/home/user/.cache/zls'
info : (server): Set config option 'enable_build_on_save' to 'true'
info : (server): Set config option 'enable_autofix' to 'true'
info : (server): Set config option 'semantic_tokens' to 'partial'
info : (server): Set config option 'warn_style' to 'true'
info : (server): Set config option 'highlight_global_var_declarations' to 'true'
```

**Comment by srijan-paul** [comment]
Interestingly, this worked on previous versions of ZLS. Appears to be a regression.

**Comment by smallkirby** [comment]
That's a strange behavior...

I attach a debug log for your information, though it seems to have no useful information: [zls-log.txt](https://github.com/user-attachments/files/17211117/zls-log.txt)

**Comment by srijan-paul** [comment]
Yeah, there'd be more logs in a debug build of zls.
I made a fix in #2043, let's see if it gets merged.
